### PR TITLE
Implement a CassandraTask for moving single-token nodes

### LINKED ISSUE #441: K8SSAND-1879 ⁃ Implement a CassandraTask for moving single-token nodes
**What is missing?**

A CassandraTask to run `nodetool move`:

https://cassandra.apache.org/doc/latest/cassandra/tools/nodetool/move.html

**Why do we need it?**

This will be useful when rebalancing unbalanced clusters.

**Anything else we need to know?**:

We will need https://github.com/k8ssandra/management-api-for-apache-cassandra/issues/219 implemented first.

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1879) by [Unito](https://www.unito.io)
┆Issue Number: K8SSAND-1879
┆Priority: Medium
┆Link To Issue: https://k8ssandra.atlassian.net/browse/K8SSAND-1879

### PR DESCRIPTION
**Pull request by adutra**
**What this PR does**:

This PR implements a CassandraTask for moving single-token nodes.

**Which issue(s) this PR fixes**:
Fixes #441 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by adutra** [comment]
Note to reviewers: I'd like to introduce the bare functionality first, then introduce later on some extra functionality such as:

* Available disk size pre-checks;
* Automatic token computation.

File: tests/testdata/single-token-three-rack-three-node-dc.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+metadata:
+  name: dc1
+spec:
+  clusterName: cluster1
+  serverType: cassandra
+  serverVersion: "4.0.1"
+  managementApiAuth:
+    insecure: {}
+  size: 3
+  serverImage: k8ssandra/cass-management-api:4.1-b6341fc
```
Comment (adutra) [inline tests/testdata/single-token-three-rack-three-node-dc.yaml]: This is temporary and needs to be removed before we merge this PR.

File: controllers/control/jobs.go [hunk 2]
```diff
@@ -211,6 +211,51 @@ func (r *CassandraTaskReconciler) replace(taskConfig *TaskConfiguration) {
 	taskConfig.PreProcessFunc = r.replacePreProcess
 }
 
+// Move functionality
+
+func callMove(nodeMgmtClient httphelper.NodeMgmtClient, pod *corev1.Pod, taskConfig *TaskConfiguration) (string, error) {
+	newToken, found := taskConfig.Arguments.NewTokens[pod.Name]
```
Comment (burmanm) [inline controllers/control/jobs.go]: Isn't this already validated in the validator phase?

Comment (adutra) [reply controllers/control/jobs.go]: Whoops yes, you are right.

**Review by burmanm — APPROVED** [review]

**Pull request merged**